lib/SPIRV/SPIRVReader.cpp

```diff
@@ -5141,36 +5141,53 @@ void SPIRVToLLVM::transAuxDataInst(SPIRVExtInst *BC) {
     return;
   auto Args = BC->getArguments();
   // Args 0 and 1 are common between attributes and metadata.
-  // 0 is the function, 1 is the name of the attribute/metadata as a string
-  auto *SpvFcn = BC->getModule()->getValue(Args[0]);
-  auto *F = static_cast<Function *>(getTranslatedValue(SpvFcn));
-  assert(F && "Function should already have been translated!");
+  // 0 is the global object, 1 is the name of the attribute/metadata as a string
+  auto *Arg0 = BC->getModule()->getValue(Args[0]);
+  auto *GO = cast<GlobalObject>(getTranslatedValue(Arg0));
+  auto *F = dyn_cast_or_null<Function>(GO);
+  auto *GV = dyn_cast_or_null<GlobalVariable>(GO);
```

Comment:
This makes me a little bit suspicious, why do we need dyn_cast**_or_null** here when `GO` is a result of a `cast`? May be it's leftover from a previous patch iteration, where some test was failing.

Reply:
done, yet it is a misuse.